README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# resume

<!-- badges: start -->

<!-- badges: end -->

{resume} provides an implementation of the [Bootstrap Resume
Template](https://github.com/BlackrockDigital/startbootstrap-resume) for
‘Shiny’.

## Installation

You can install the dev version of `{resume}` from GitHub with:

``` r
remotes::install_github("ColinFay/resume")
```

## Using {resume}

A `resume_page` takes 3 elements:

  - `app_title`: title of the application
  - `nav`: a navigation pane, created by `resume_navbar`. This function
    has two arguments: `refs`, a named list of characters, with names
    being ID of sections, and elements being text displayed, `image`,
    which is a link to an image to display on top of the app.
  - `body`: the body of the app, built with `resume_body` &
    `resume_section`. For each element in `resume_navbar/refs`, there
    should be one `resume_section`. The link between the two is made
    through the ID (name of `resume_navbar/refs` ==
    `resume_section/id`).

## Example

Example app:

``` r
library(shiny)
library(resume)
ui <- function(request){
  resume_page(
    app_title = "plop",
    nav = resume_navbar(
      refs = c(
        "about" = "About",
        "experience" = "Experience",
        "education" = "Education",
        "interests" = "Interests",
        "awards" = "Awards"
      ),
      image = "https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/apple/198/sign-of-the-horns_1f918.png", 
      color = "pink"
    ),
    body = resume_body(
      # About
      resume_section(
        id = "about",
        h4('about'),
        plotOutput("about")
      ),
      # experience
      resume_section(
        id = "experience",
        h4('experience'),
        plotOutput("experience")
      ),
      # education
      resume_section(
        id = "education",
        h4('education'),
        plotOutput("education")
      ),
      # interests
      resume_section(
        id = "interests",
        h4('interests'),
        plotOutput("interests")
      ),
      # awards
      resume_section(
        id = "awards",
        h4('awards'),
        plotOutput("awards")
      )
    )
  )
}

server <- function(input, output, session){
  output$about <- renderPlot({
    plot(iris)
  })
  output$experience <- renderPlot({
    plot(mtcars)
  })
  output$education <- renderPlot({
    plot(airquality)
  })
  output$interests <- renderPlot({
    plot(AirPassengers)
  })
  output$awards <- renderPlot({
     plot(anscombe)
  })
}

shinyApp(ui, server)
```

![](readme-figs/example.gif)

Please note that the ‘resume’ project is released with a [Contributor
Code of Conduct](CODE_OF_CONDUCT.md). By contributing to this project,
you agree to abide by its terms.


NEWS.md
# resume 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.


R/nav.R
#' @importFrom htmltools tags
make_navbar <- function(
  nav
){
  if (is.null(names(nav))){
    names(nav) <- nav
  }
  mapply(function(x, y){
    tags$li(
      class = "nav-item",
      tags$a(
        class = "nav-link js-scroll-trigger",
        href = sprintf("#%s", x),
        y
      )
    )
  },
  names(nav),
  nav,
  SIMPLIFY = FALSE
  )
}

#' A Resume Navbar
#'
#' @param refs a named list with names being the ID and Elements being the text displayed.
#' @param color Color of the sidebar
#' @param image Image to put in the nav.
#'
#' @return An HTML taglist
#' @export
#'
#' @importFrom htmltools tags
resume_navbar <- function(
  refs = c("a" = "A"),
  image = "img/profile.jpg",
  color = "#bd5d38"
){
  tagList(
    tags$style(
      sprintf(".navbar { background-color: %s!important;}", color)
    ),
    tags$nav(class = "navbar navbar-expand-lg navbar-dark bg-primary fixed-top",
             id = "sideNav",
             tags$a(class = "navbar-brand js-scroll-trigger",
                    href = "#page-top",
                    tags$span(class = "d-block d-lg-none",
                              "nav_title"),
                    tags$span(class = "d-none d-lg-block",
                              tags$img(class = "img-fluid img-profile rounded-circle mx-auto mb-2",
                                       src = image,
                                       alt = NA))),
             tags$button(class = "navbar-toggler",
                         type = "button",
                         "data-toggle" = "collapse",
                         "data-target" = "#navbarSupportedContent",
                         "aria-controls" = "navbarSupportedContent",
                         "aria-expanded" = "false",
                         "aria-label" = "Toggle navigation",
                         tags$span(class = "navbar-toggler-icon")),
             tags$div(class = "collapse navbar-collapse",
                      id = "navbarSupportedContent",
                      tags$ul(class = "navbar-nav",
                              make_navbar(
                                refs
                              )
                      )
             )
    )
  )
}


R/page.R
#' Resume page
#'
#' A Shinny App with the Bootstrap Resume Template
#'
#' @param app_title Title of the App
#' @param nav A navbar, constructed with \code{resume_navbar}
#' @param body A body, constructed with \code{resume_body} & \code{resume_sections}
#' @importFrom htmltools tagList tags
#' @return An HTML taglist
#' @export
resume_page <- function(
  app_title,
  nav,
  body
){
  tagList(
    tags$html(
      resume_deps(),
      resume_head(title = app_title),
      tags$body(
        id="page-top",
        nav,
        body
      )
    )
  )
}

#' Body of the page
#'
#' @param ... A series of \code{resume_section}
#'
#' @return An HTML taglist
#' @export
#' @importFrom htmltools tagList tags
resume_body <- function(...){
  tagList(
    tags$div(
      class="container-fluid p-0",
      ...,
      resume_scripts()
    )
  )
}

#' A section in the page
#'
#' @param id ID of the sections, should reflect the \code{resume_navbar} structure
#' @param ... elements to put inside the section
#'
#' @return An HTML taglist
#' @export
#' @importFrom htmltools tagList tags
resume_section <- function(id, ...){
  tagList(
    tags$section(
      class="resume-section p-3 p-lg-5 d-flex align-items-center",
      id = id,
      tags$div(
        class="w-100",
        ...
      )
    )
  )
}


R/resume_deps.R
#' @importFrom htmltools htmlDependency
resume_deps <- function(){
  htmlDependency(
    "resume", "5.0.6",
    src = system.file("startbootstrap-resume-gh-pages/", package = "resume"),
    stylesheet = list.files(
      system.file("startbootstrap-resume-gh-pages/", package = "resume"),
      pattern = "\\.css$", recursive = TRUE
    ),
    script = list.files(
      system.file("startbootstrap-resume-gh-pages/", package = "resume"),
      pattern = "\\.js$", recursive = TRUE
    )
  )
}

#' @importFrom htmltools HTML
resume_head <- function(
  title = "Resume - Start Bootstrap Theme"
){
  HTML(
    sprintf(
      '<head>

    <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="">
  <meta name="author" content="">

  <title>%s</title>

  <!-- Bootstrap core CSS -->
  <link href="resume-5.0.6/vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">

  <!-- Custom fonts for this template -->
  <link href="https://fonts.googleapis.com/css?family=Saira+Extra+Condensed:500,700" rel="stylesheet">
  <link href="https://fonts.googleapis.com/css?family=Muli:400,400i,800,800i" rel="stylesheet">
  <link href="resume-5.0.6/vendor/fontawesome-free/css/all.min.css" rel="stylesheet">

  <!-- Custom styles for this template -->
  <link href="resume-5.0.6/css/resume.min.css" rel="stylesheet">

</head>', title
    )
  )
}

#' @importFrom htmltools HTML
resume_scripts <- function(){
  HTML(
    '  <!-- Bootstrap core JavaScript -->
  <script src="resume-5.0.6/vendor/jquery/jquery.min.js"></script>
  <script src="resume-5.0.6/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>

  <!-- Plugin JavaScript -->
  <script src="resume-5.0.6/vendor/jquery-easing/jquery.easing.min.js"></script>

  <!-- Custom scripts for this template -->
  <script src="resume-5.0.6/js/resume.min.js"></script>'
  )
}


inst/examples/101.R
library(shiny)
ui <- function(request){
  resume_page(
    app_title = "plop",
    nav = resume_navbar(
      refs = c(
        "about" = "About",
        "experience" = "Experience",
        "education" = "Education",
        "interests" = "Interests",
        "awards" = "Awards"
      ),
      nav_title = "plop",
      image = "https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/apple/198/sign-of-the-horns_1f918.png"
    ),
    body = resume_body(
      # About
      resume_section(
        id = "about",
        h4('about'),
        plotOutput("about")
      ),
      # experience
      resume_section(
        id = "experience",
        h4('experience'),
        plotOutput("experience")
      ),
      # education
      resume_section(
        id = "education",
        h4('education'),
        plotOutput("education")
      ),
      # interests
      resume_section(
        id = "interests",
        h4('interests'),
        plotOutput("interests")
      ),
      # awards
      resume_section(
        id = "awards",
        h4('awards'),
        plotOutput("awards")
      )
    )
  )
}

server <- function(input, output, session){
  output$about <- renderPlot({
    plot(iris)
  })
  output$experience <- renderPlot({
    plot(iris)
  })
  output$education <- renderPlot({
    plot(iris)
  })
  output$interests <- renderPlot({
    plot(iris)
  })
  output$awards <- renderPlot({
    plot(iris)
  })
}

shinyApp(ui, server)
